examples/js_binary/BUILD.bazel

```diff
@@ -1,18 +1,26 @@
+load("@acorn__8.4.0__links//:link_js_package.bzl", link_acorn = "link_js_package")
 load("@aspect_bazel_lib//lib:directory_path.bzl", "directory_path")
 load("@bazel_skylib//rules:diff_test.bzl", "diff_test")
 load("@bazel_skylib//rules:write_file.bzl", "write_file")
-load("@npm//:defs.bzl", "package", "package_dir")
-load("//js:defs.bzl", "js_binary", "js_test")
+load("@npm//examples/npm_deps/@aspect-test/a:package_json.bzl", aspect_test_a_bin = "bin")
+load("//js:defs.bzl", "js_binary", "js_test", "link_js_package")
 load("//js:run_js_binary.bzl", "run_js_binary")
 
+# Link the acorn package, which was fetched separately with npm_import from /WORKSPACE, as
+# a direct dependency in bazel-bin/examples/js_binary/node_modules
+link_acorn(name = "acorn_link")
+
+# Link @mycorp/mylib as a direct dependency in bazel-bin/examples/js_binary/node_modules
+link_js_package(name = "mycorp_mylib_link")
+
 ####################################################
 # Test case 4
 # Show that a js_binary can be used with run_js_binary
 # because everything it needs to run is in the runfiles
 
 js_binary(
     name = "bin",
-    data = [package("acorn", "examples")],
+    data = [":acorn_link"],
```

Comment:
yeah this is way better just to refer to the target above, feels like Bazel idiom again